besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/liveness/LivenessPlugin.java

```diff
@@ -0,0 +1,429 @@
+  }
+
+  /**
+   * Gets the plugin configuration.
+   *
+   * @return The LivenessPluginConfiguration
+   */
+  private LivenessPluginConfiguration getConfiguration() {
+    return livenessPluginConfiguration();
+  }
+
+  @Override
+  public void onBlockAdded(AddedBlockContext addedBlockContext) {
+    if (!isPluginEnabled) return;
+
+    BlockHeader newBlock = addedBlockContext.getBlockHeader();
+    lastProcessedBlock.set(newBlock);
+
+    // Reset the last reported timestamp when a new block is added
+    lastReportedTimestamp.set(newBlock.getTimestamp());
+
+    log.debug(
+        "New block added: lastProcessedBlock ={}, lastReportedTimestamp ={}",
+        lastProcessedBlock.get().getNumber(),
+        lastReportedTimestamp.get());
+  }
+
+  /**
+   * Checks the timestamp of the last block and reports downtime if necessary. This method is called
+   * periodically by the scheduler.
+   */
+  private void checkBlockTimestampAndReport() {
+    if (!isPluginEnabled) return;
+
+    try {
+      BlockHeader lastBlock = lastProcessedBlock.get();
+      if (lastBlock == null) {
+        log.warn("No blocks available in the blockchain");
+        return;
+      }
+
+      long currentTimestamp = Instant.now().getEpochSecond();
+      long lastBlockTimestamp = lastBlock.getTimestamp();
+      long timeSinceLastBlock = currentTimestamp - lastBlockTimestamp;
+
+      log.debug(
+          "Checking block timestamp: lastBlockNumber={}, lastBlockTimestamp={}, currentTimestamp={}, timeSinceLastBlock={}s",
+          lastBlock.getNumber(),
+          lastBlockTimestamp,
+          currentTimestamp,
+          timeSinceLastBlock);
+
+      // Check if we need to report downtime
+      if (timeSinceLastBlock > livenessPluginConfiguration.maxBlockAgeMilliseconds()) {
+        // Only report if we haven't reported recently or if significant time has passed
+        long timeSinceLastReport = currentTimestamp - lastReportedTimestamp.get();
+        if (timeSinceLastReport > livenessPluginConfiguration.maxBlockAgeMilliseconds()) {
+          log.info(
+              "Sequencer appears to have been down: lastBlockNumber={}, lastBlockTimestamp={}, timeSinceLastBlock={}s",
+              lastBlock.getNumber(),
+              lastBlockTimestamp,
+              timeSinceLastBlock);
+
+          sendSequencerUptimeTransaction(lastBlockTimestamp, currentTimestamp);
+          lastReportedTimestamp.set(currentTimestamp);
+
+          // Update lastProcessedBlock to current chain head to avoid re-triggering
+          BlockHeader currentBlock = blockchainService.getChainHeadHeader();
+          if (currentBlock != null) {
+            lastProcessedBlock.set(currentBlock);
+            log.debug(
+                "Updated lastProcessedBlock to current chain head: {}", currentBlock.getNumber());
+          }
+
+          if (uptimeTransactionsCounter != null) {
+            uptimeTransactionsCounter.inc(2); // We send two transactions (legacy counter)
```

Comment:
I believe it's possible to get here with 1 or more transactions failing to send. So the metrics counter will go up by 2 but actual transactions sent was 0, 1 or 2

Should we couple the metric updates with successful confirmation of the transactions?